### ISSUE TITLE
`wtf` cannot handle worktrees with directories have the same name as the current branch

### ISSUE THREAD

**Issue by mih** [datalad/datalad#6650]
Demo:

```
mih@meiner ~/hacking/datalad/next (git)-[docs] % dl wtf
CommandError: 'git -c diff.ignoreSubmodules=none rev-list docs' failed with exitcode 128
fatal: ambiguous argument 'docs': both revision and filename
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

Accidentally discovered after posting a PR from a `docs` branch.

**Comment by yarikoptic** [comment]
I think it could be resolved by adding explicit `--` at the end (assuming that `docs` is the branch)

**Comment by adswa** [comment]
was fixed with #6661 if I am not mistaken.
